lib/trino-orc/src/main/java/io/trino/orc/DeflateCompressor.java

```diff
@@ -58,8 +58,8 @@ public int compress(byte[] input, int inputOffset, int inputLength, byte[] outpu
     }
 
     @Override
-    public void compress(ByteBuffer input, ByteBuffer output)
+    public int compress(MemorySegment input, MemorySegment output)
     {
-        throw new UnsupportedOperationException("Compression of byte buffer not supported for deflate");
```

Comment:
We shuld be able to drop this ckass  since I added deflate directly to Aircompressor https://github.com/airlift/aircompressor/commit/6eb421bbd3590c2a0887418e5c5ca4a389a4887d